server/src/utils/unreal.rs

```diff
@@ -0,0 +1,78 @@
+use symbolic::unreal::{Unreal4Crash, Unreal4Error, Unreal4FileType};
+
+//use semaphore_general::protocol::Event;
+//use semaphore_general::types::Annotated;
+
+use crate::constants::{ITEM_NAME_BREADCRUMBS1, ITEM_NAME_BREADCRUMBS2, ITEM_NAME_EVENT};
+use crate::envelope::{AttachmentType, ContentType, Envelope, Item, ItemType};
+
+pub fn expand_if_unreal_envelope(envelope: &mut Envelope) -> Result<&mut Envelope, Unreal4Error> {
```

Comment:
There's little point in returning the mutable reference here. However, it might make sense to return the `Event` object, which can then be used by `EventProcessor::extract_event`.

Reply:
Yes, I'm still not 100% done with the signature of the function.
Initially I wanted something like  `fn f ( env: Envelope)->Result<Envelope, Err>` to express more clearly that the function might create a new envelope (although in practice I reuse the shell of the envelope, everything inside is consumed and new items inserted).